Automattic/jetpack
# [RNMobile] Sync VideoPress metadata when post is manually saved

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5666

## Related PRs

* `Gutenberg`: https://github.com/WordPress/gutenberg/pull/49781
* `Gutenberg Mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5663
* `iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/20522
* `Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/18279

## Proposed changes:

* https://github.com/Automattic/jetpack/pull/30131/commits/43ba4149891c773d7f8be6685828138510d58e3a: The temporary "Save Settings" button (added in https://github.com/Automattic/jetpack/pull/29997) has been removed. 
* https://github.com/Automattic/jetpack/pull/30131/commits/85881662851ba106ac6969414bb135fe13497abf: A listener has been set up to listener for "save events" that come when a user manually saves a post in the native apps. Data will now only be synced from the app to the web when a save event has been detected.

### Other information:


## Jetpack product discussion

p9ugOq-3mV-p2

## Does this pull request change what data or activity we track or use?

No, it doesn't.

## Testing instructions:

* In the app, either add a new VideoPress block or navigate to an existing one.
* Tweak the block's settings. It's particularly important to change the settings in the **Details** and **Privacy and Rating** panels, as these are the settings that are not currently being synced as expected.
* Manually save your changes in the app. You can find different flows for manually saving on both Android and iOS below.
* Confirm that you see the expected logs in the console, showing that the settings have been synced.
* Navigate to the block on the web and verify that the block's settings have been correctly synced there.
* Go through a few tries of editing the block on the app then returning the web, and vice versa, to ensure syncing is working as expected.

### Android flows

<details>
 <summary>Flows for saving a draft ⤵️</summary>

* Tap the three dots to the upper right and then **Save**.
* Tap the back button.
* Tap the **Publish** button.
* The the three dots to the upper right to get to **Post Settings**.  From here, set the post's publish date and then **Schedule**.

</details>

<details>
 <summary>Flows for saving a published post ⤵️</summary>

* Tap the **Update** button.
* Tap the back button.
* The the three dots to the upper right to get to **Post Settings**.  From here, change the post's publish date and then **Schedule**.</details>

<details>
 <summary>Flows for saving a scheduled post ⤵️</summary>

* Tap the **Schedule** button.
* Tap the back button.
* The the three dots to the upper right to get to **Post Settings**.  From here, change the post's publish date and then **Schedule**.

</details>

### iOS flows

<details>
 <summary>Flows for saving a draft ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Draft**.
* Tap the three dots to the upper right and then **Publish**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>

<details>
 <summary>Flows for saving a published post ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Post**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>

<details>
 <summary>Flows for saving a scheduled post ⤵️</summary>

* Tap the **Update** button.
* Tap the X icon to exit the post and **Update Post**.
* The the three dots to the upper right to get to Post Settings. From here, set the post's publish date and then **Schedule**.

</details>
